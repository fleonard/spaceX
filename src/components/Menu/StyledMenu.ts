import styled from "styled-components/macro";

import { Box, List } from "@material-ui/core";

interface MenuProps {
  readonly open: boolean;
}

const StyledMenuContainer = styled(Box)<MenuProps>`
  background: #000;
  color: #fff;
  height: calc(100vh - 60px);
  padding: 20px;
  position: absolute;
  right: 0;
  top: 60px;
  transform: ${({ open }) => (open ? "translateX(0)" : "translateX(100%)")};
  transition: transform 0.3s ease-in-out;
  width: 270px;

  @media (min-width: 768px) {
    display: none;
  }
`;

const StyledMenu = styled(List)`
  display: flex;
  flex-direction: column;
  padding: 0;
  text-align: right;
`;

const MenuItem = styled(Box)`
  display: flex;
  padding-top: 10px;
  width: 100%;

  a {
    border-bottom: 1pt solid #252525;
    color: #fff;
    padding: 10px 0;
    text-decoration: none;
    text-transform: uppercase;
    width: 100%;
  }
`;

export { StyledMenuContainer, StyledMenu, MenuItem };
