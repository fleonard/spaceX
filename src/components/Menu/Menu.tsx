import React from "react";
import { NavLink } from "react-router-dom";

import { StyledMenuContainer, StyledMenu, MenuItem } from "./StyledMenu";

interface MenuProps {
  open: boolean;
  id: string;
}

const Menu: React.FunctionComponent<MenuProps> = ({ open, ...props }) => {
  const isHidden = open ? true : false;

  return (
    <StyledMenuContainer open={open} aria-hidden={!isHidden} {...props}>
      <StyledMenu>
        <MenuItem component="li">
          <NavLink to="/rockets">Rockets</NavLink>
        </MenuItem>
        <MenuItem component="li">
          <NavLink to="/dragons">Dragons</NavLink>
        </MenuItem>
      </StyledMenu>
    </StyledMenuContainer>
  );
};

export default Menu;
