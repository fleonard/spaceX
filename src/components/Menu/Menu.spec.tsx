import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import "jest-styled-components";

import Menu from "./Menu";

describe("Menu Component", () => {
  it("renders when closed", () => {
    const { container } = render(
      <BrowserRouter>
        <Menu open={false} id="mobile-menu" />
      </BrowserRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders when open", () => {
    const { container } = render(
      <BrowserRouter>
        <Menu open={true} id="mobile-menu" />
      </BrowserRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
