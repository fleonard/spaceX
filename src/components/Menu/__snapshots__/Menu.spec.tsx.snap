// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component renders when closed 1`] = `
.c0 {
  background: #000;
  color: #fff;
  height: calc(100vh - 60px);
  padding: 20px;
  position: absolute;
  left: 0;
  top: 60px;
  -webkit-transform: translateX(-100%) translateZ(0);
  -ms-transform: translateX(-100%) translateZ(0);
  transform: translateX(-100%) translateZ(0);
  -webkit-transform-origin: 0% 0%;
  -ms-transform-origin: 0% 0%;
  transform-origin: 0% 0%;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  width: 270px;
  z-index: 10;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  text-align: right;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 10px;
  width: 100%;
}

.c2 a {
  border-bottom: 1pt solid #252525;
  color: #fff;
  padding: 10px 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  width: 100%;
}

@media (min-width:768px) {
  .c0 {
    display: none;
  }
}

<div
  aria-hidden="true"
  class="MuiBox-root MuiBox-root-1 c0"
  id="mobile-menu"
>
  <ul
    class="MuiList-root c1 MuiList-padding"
  >
    <li
      class="MuiBox-root MuiBox-root-2 c2"
    >
      <a
        href="/rockets"
      >
        Rockets
      </a>
    </li>
    <li
      class="MuiBox-root MuiBox-root-3 c2"
    >
      <a
        href="/dragons"
      >
        Dragons
      </a>
    </li>
  </ul>
</div>
`;

exports[`Menu Component renders when open 1`] = `
.c0 {
  background: #000;
  color: #fff;
  height: calc(100vh - 60px);
  padding: 20px;
  position: absolute;
  left: 0;
  top: 60px;
  -webkit-transform: translateX(0) translateZ(0);
  -ms-transform: translateX(0) translateZ(0);
  transform: translateX(0) translateZ(0);
  -webkit-transform-origin: 0% 0%;
  -ms-transform-origin: 0% 0%;
  transform-origin: 0% 0%;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  width: 270px;
  z-index: 10;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  text-align: right;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 10px;
  width: 100%;
}

.c2 a {
  border-bottom: 1pt solid #252525;
  color: #fff;
  padding: 10px 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  width: 100%;
}

@media (min-width:768px) {
  .c0 {
    display: none;
  }
}

<div
  aria-hidden="false"
  class="MuiBox-root MuiBox-root-4 c0"
  id="mobile-menu"
  open=""
>
  <ul
    class="MuiList-root c1 MuiList-padding"
  >
    <li
      class="MuiBox-root MuiBox-root-5 c2"
    >
      <a
        href="/rockets"
      >
        Rockets
      </a>
    </li>
    <li
      class="MuiBox-root MuiBox-root-6 c2"
    >
      <a
        href="/dragons"
      >
        Dragons
      </a>
    </li>
  </ul>
</div>
`;
